omf/compat/omf_v1.py

```diff
@@ -0,0 +1,456 @@
+        self._filename = filename
+
+    def load(self, include_binary: bool = True, project_json: bool = None):
+        self._include_binary = include_binary
+        try:
+            with open(self._filename, 'rb') as self._f:
+                project_uuid, json_start = self._read_header()
+                self._project = self._read_json(json_start)
+                try:
+                    return self._convert_project(project_uuid)
+                except properties.ValidationError as exc:
+                    raise InvalidOMFFile(exc)
+        finally:
+            self._reset()
+
+    def _reset(self):
+        self._f = None
+        self._project = None
+        self._include_binary = True
+        self._attribute_bucket = dict()
+        self.__cache = {}  # uuid -> reusable item
+
+    def _read_header(self):
+        """Checks magic number and version; gets project uid and json start"""
+        self._f.seek(0)
+        if self._f.read(4) != b'\x84\x83\x82\x81':
+            raise WrongVersionError(f'Unsupported format: {self._filename}')
+        file_version = struct.unpack('<32s', self._f.read(32))[0]
+        file_version = file_version[0:len(COMPATIBILITY_VERSION)]
+        if file_version != COMPATIBILITY_VERSION:
+            raise WrongVersionError("Unsupported file version: {}".format(file_version))
+        project_uuid = uuid.UUID(bytes=struct.unpack('<16s', self._f.read(16))[0])
+        json_start = struct.unpack('<Q', self._f.read(8))[0]
+        return str(project_uuid), json_start
+
+    def _read_json(self, json_start):
+        self._f.seek(json_start, 0)
+        return json.loads(self._f.read().decode('utf-8'))
+
+    # Safe access to attributes
+    @classmethod
+    def __get_attr(cls, src, attr, optional=False, default=_default):
+        try:
+            value = src[attr]
+        except TypeError:
+            raise InvalidOMFFile(f"Attribute {attr} missing")
+        except KeyError:
+            if not optional:
+                raise InvalidOMFFile(f"Attribute {attr} missing")
+            if default is _default:
+                return None
+            value = default
+        return value
+
+    @classmethod
+    def __require_attr(cls, src, attr, required_value):
+        value = cls.__get_attr(src, attr)
+        if value != required_value:
+            raise InvalidOMFFile(f"Invalid attribute {attr}. Expected: {required_value}, actual: {value}")
+
+    @classmethod
+    def __copy_attr(cls, src, src_attr, dst, dst_attr=None,
+                    optional_src=False, optional_dst=False, default=_default):
+        if dst_attr is None:
+            dst_attr = src_attr
+        try:
+            value = src[src_attr]
+        except TypeError:
+            raise InvalidOMFFile(f"Attribute {src_attr} missing")
+        except KeyError:
+            if not optional_src:
+                raise InvalidOMFFile(f"Attribute {src_attr} missing")
+            if default is _default:
+                return
```

Comment:
From `try:` down this seems identical to `__get_attr `above ^^^ can we not just call that?

Reply:
I thought about it but `__copy_attr` has a `return` in the `KeyError` branch. `__get_attr` treats this branch as "return the default value of None" - while `__copy_attr` treats this as "don't set the attr at all". If I call `__get_attr` here, I can't tell if the `None` is from this branch or the actual return value